src/bubble-chart/bubble-series.component.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  Component,
+  Input,
+  Output,
+  SimpleChanges,
+  EventEmitter,
+  OnChanges,
+  ChangeDetectionStrategy,
+  trigger,
+  style,
+  transition,
+  animate
+} from '@angular/core';
+import { Location } from '@angular/common';
+import { formatLabel } from '../common/label.helper';
+import { id } from '../utils/id';
+
+@Component({
+  selector: 'g[ngx-charts-bubble-series]',
+  template: `
+    <svg:g *ngFor="let circle of circles">
+      <svg:g ngx-charts-circle
+        *ngIf="isVisible(circle)"
+        class="circle"
+        [cx]="circle.cx"
+        [cy]="circle.cy"
+        [r]="circle.radius"
+        [fill]="circle.color"
+        [class.active]="isActive({name: circle.seriesName})"
+        [pointerEvents]="circle.value === 0 ? 'none': 'all'"
+        [data]="circle.value"
+        [classNames]="circle.classNames"
+        (select)="onClick($event, circle.label)"
+        (activate)="activateCircle(circle)"
+        (deactivate)="deactivateCircle(circle)"
+        ngx-tooltip
+        [tooltipPlacement]="'top'"
+        [tooltipType]="'tooltip'"
+        [tooltipTitle]="getTooltipText(circle)"
+      />
+    </svg:g>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('animationState', [
+      transition('void => *', [
+        style({
+          opacity: 0,
+        }),
+        animate(250, style({opacity: 1}))
+      ])
+    ])
+  ]
+})
+export class BubbleSeriesComponent implements OnChanges {
+
+  @Input() data;
+  @Input() type = 'standard';
+  @Input() xScale;
+  @Input() yScale;
+  @Input() rScale;
+  @Input() colors;
+  @Input() visibleValue;
+  @Input() activeEntries: any[];
+
+  @Output() select = new EventEmitter();
+  @Output() activate = new EventEmitter();
+  @Output() deactivate = new EventEmitter();
+
+  areaPath: any;
+  circles: any[];
+
+  constructor(private location: Location) {
```

Comment:
We use Location for gradients. Since we removed the gradient, we don't need this any more.